"""Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT."""

from __future__ import annotations
import dataclasses
from dataclasses_json import Undefined, dataclass_json
from enum import Enum
from sdk import utils
from typing import Optional

class ControlErrorResponseStatusEnum(str, Enum):
    r"""Code | Text | Description
    -- | -- | --
    0 | Success | The request was successfully accepted by Vonage.
    1 | Throttled | You are trying to send more than the maximum of 30 requests per second.
    2 | Your request is incomplete and missing the mandatory parameter `$parameter` | The stated parameter is missing.
    3 | Invalid value for parameter `$parameter` | Invalid value for parameter. If you see Facility not allowed in the error text, check that you are using the correct Base URL in your request.
    4 | Invalid credentials were provided | The supplied API key or secret in the request is either invalid or disabled.
    5 | Internal Error | An error occurred processing this request in the Cloud Communications Platform.
    6 | The Vonage platform was unable to process this message for the following reason: `$reason` | The request could not be routed.
    8 | The api_key you supplied is for an account that has been barred from submitting messages. |
    9 | Partner quota exceeded | Your account does not have sufficient credit to process this request.
    19 | For `cancel`: Either you have not waited at least 30 secs after sending a Verify request before cancelling or Verify has made too many attempts to deliver the verification code for this request and you must now wait for the process to complete. For `trigger_next_event`: All attempts to deliver the verification code for this request have completed and there are no remaining events to advance to.
    """
    ZERO = '0'
    ONE = '1'
    TWO = '2'
    THREE = '3'
    FOUR = '4'
    FIVE = '5'
    SIX = '6'
    EIGHT = '8'
    NINE = '9'
    NINETEEN = '19'


@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class ControlErrorResponse:
    r"""Error"""
    
    error_text: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.get_field_name('error_text'), 'exclude': lambda f: f is None }})
    r"""If the `status` is non-zero, this explains the error encountered."""  
    status: Optional[ControlErrorResponseStatusEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.get_field_name('status'), 'exclude': lambda f: f is None }})
    r"""Code | Text | Description
    -- | -- | --
    0 | Success | The request was successfully accepted by Vonage.
    1 | Throttled | You are trying to send more than the maximum of 30 requests per second.
    2 | Your request is incomplete and missing the mandatory parameter `$parameter` | The stated parameter is missing.
    3 | Invalid value for parameter `$parameter` | Invalid value for parameter. If you see Facility not allowed in the error text, check that you are using the correct Base URL in your request.
    4 | Invalid credentials were provided | The supplied API key or secret in the request is either invalid or disabled.
    5 | Internal Error | An error occurred processing this request in the Cloud Communications Platform.
    6 | The Vonage platform was unable to process this message for the following reason: `$reason` | The request could not be routed.
    8 | The api_key you supplied is for an account that has been barred from submitting messages. |
    9 | Partner quota exceeded | Your account does not have sufficient credit to process this request.
    19 | For `cancel`: Either you have not waited at least 30 secs after sending a Verify request before cancelling or Verify has made too many attempts to deliver the verification code for this request and you must now wait for the process to complete. For `trigger_next_event`: All attempts to deliver the verification code for this request have completed and there are no remaining events to advance to.
    """  
    